server/backends/blobstore/blobstore.go

```diff
@@ -193,7 +293,9 @@ func (g *GCSBlobStore) ReadBlob(ctx context.Context, blobName string) ([]byte, e
 		}
 		return nil, err
 	}
-	return decompress(ioutil.ReadAll(reader))
+	return decompress(trackRead(gcsLabel, func() ([]byte, error) {
+		return ioutil.ReadAll(reader)
+	}))
```

Comment:
What do you think about deferring your tracking function from inside these methods, rather than wrapping the read/write/delete functions in your own?

I wrote out what I think it would look like and it's not too bad -- I think it has the advantage of being a little clearer to read and it's more obvious that a tracking error won't cause a read error.

For example, instead of:

	return decompress(trackRead(gcsLabel, func() ([]byte, error) {
		return ioutil.ReadAll(reader)
	}))

We could do:
	start := time.Now()
	n, err := ioutil.ReadAll(reader)
	defer trackRead(start, n, err)
	return decompress(n, err)

Reply:
That SG, did you mean something like this so we don't track the decompression time (seems like we'd want to measure that separately)?

```go
func ReadBlob(...) {
  start := time.Now()
  n, err := downloadFromGCS(...) // New method
  trackRead("gcs", start, n, err)
  return decompress(n, err)
}
```

Reply:
Rewrote with the `defer` omitted, lmk if you also think decompression time should be included. I was thinking we might add decompression time to internal metrics in the future if, for example, a new compression library comes out and we want to compare before & after.